src/ocl/convolutionocl.cpp

```diff
@@ -578,6 +579,56 @@ void ValidateAlphaBeta(const void* alpha, const void* beta)
     }
 }
 
+void DumpTensorToFileFromDevice(const miopen::Handle& handle,
+                                const miopen::TensorDescriptor& tDesc,
+                                ConstData_t dData,
+                                const std::string& filename)
+{
+    if(dData == nullptr)
+    {
+        MIOPEN_LOG_E("Dereferencing nullptr when trying to dump tensor from gpu");
+        return;
+    }
+    namespace fs = boost::filesystem;
+
+    fs::path file_name_with_path(filename);
+    fs::path path = file_name_with_path.parent_path();
+
+    // dump to current folder if full path not provided.
+    if(path.empty())
+    {
+        path                = fs::current_path();
+        file_name_with_path = path / file_name_with_path; // append paths
+    }
+    if(!fs::exists(path))
+    {
+        MIOPEN_LOG_E("Directory does not exists : " << path);
+        return;
+    }
+    std::string file_name_with_path_str = file_name_with_path.string();
+
+    std::ofstream file_stream;
+    file_stream.open(file_name_with_path_str);
+
+    if(!file_stream.is_open())
+    {
+        MIOPEN_LOG_E("Cannot write to file : " << file_name_with_path_str);
+        return;
+    }
+
+    // read tensor data from gpu
+    size_t num_bytes = tDesc.GetNumBytes();
+    MIOPEN_LOG_I2("Start bringing tensor from device to host");
```

Comment:
I highly recommend moving this to Trace level or removing this logging.

Reply:
I will remove it.